src/app/portlets/content-types/main/content-types.component.ts

```diff
@@ -1,119 +1,110 @@
+        'contenttypes.content.form',
+        'contenttypes.confirm.message.delete',
+        'contenttypes.confirm.message.delete.content',
+        'contenttypes.confirm.message.delete.warning',
+        'contenttypes.action.delete',
+        'contenttypes.action.cancel',
+        'Content-Type'
+    ];
     @ViewChild('listing') listing: ListingDataTableComponent;
 
     constructor(
-        messageService: MessageService,
+        private messageService: MessageService,
         private router: Router,
         private route: ActivatedRoute,
         private contentTypesInfoService: ContentTypesInfoService,
         private crudService: CrudService,
-        private dotConfirmationService: DotConfirmationService
-    ) {
-        super(
-            [
-                'contenttypes.fieldname.structure.name',
-                'contenttypes.content.variable',
-                'contenttypes.form.label.description',
-                'contenttypes.fieldname.entries',
-                'message.structure.delete.structure.and.content',
-                'message.structure.cantdelete',
-                'contenttypes.content.file',
-                'contenttypes.content.content',
-                'contenttypes.content.persona',
-                'contenttypes.content.widget',
-                'contenttypes.content.page',
-                'contenttypes.confirm.message.delete',
-                'contenttypes.confirm.message.delete.content',
-                'contenttypes.confirm.message.delete.warning',
-                'contenttypes.action.delete',
-                'contenttypes.action.cancel',
-                'Content-Type'
-            ],
-            messageService
-        );
+        private dotConfirmationService: DotConfirmationService,
+        private dotContentletService: DotContentletService
+    ) {}
+
+    ngOnInit() {
+        Observable.forkJoin(
+            this.messageService.getMessages(this.i18nKeys),
+            this.dotContentletService.getAllContentTypes()
+        ).subscribe(val => {
+            this.i18nMessages = val[0];
+            this.actionHeaderOptions = {
+                primary: {
+                    command: $event => {
+                        this.createContentType($event);
+                    },
+                    model: this.setContentTypes(val[1])
+                }
+            };
+            this.contentTypeColumns = this.setContentTypeColumns();
+            this.rowActions = [
+                {
+                    label: 'Remove',
+                    icon: 'fa-trash',
+                    command: item => this.removeConfirmation(item)
+                }
+            ];
+        });
     }
 
-    /**
-     * Callback call from BaseComponent when the messages are received.
-     * @memberOf ContentTypesPortletComponent
-     */
-    onMessage(): void {
-        this.actionHeaderOptions = {
-            primary: {
+    private setContentTypes(arr: StructureTypeView[]): ButtonModel[] {
```

Comment:
Please use sematyc name, ```arr``` could be ```baseTypes```

(I know the data datatype is ```StructureTypeView[]``` but we have to change that interface name, it's a old code```